driver-core/src/main/com/mongodb/client/model/expressions/DocumentExpression.java

```diff
@@ -16,10 +16,77 @@
 
 package com.mongodb.client.model.expressions;
 
+import org.bson.conversions.Bson;
+
+import java.time.Instant;
+
+import static com.mongodb.client.model.expressions.Expressions.of;
+
 /**
  * Expresses a document value. A document is an ordered set of fields, where the
  * key is a string value, mapping to a value of any other expression type.
  */
 public interface DocumentExpression extends Expression {
 
+    DocumentExpression setField(String path, Expression exp);
```

Comment:
According to the setField documentation:

> [$setField](https://www.mongodb.com/docs/manual/reference/operator/aggregation/setField/#mongodb-expression-exp.-setField) doesn't implicitly traverse objects or arrays. For example, [$setField](https://www.mongodb.com/docs/manual/reference/operator/aggregation/setField/#mongodb-expression-exp.-setField) evaluates a field value of "a.b.c" as a top-level field "a.b.c" instead of as a nested field, { "a": { "b": { "c": } } }.

So it's not correct to call the first parameter `path`.  It's really the field name.

Reply:
Updated to `fieldName`. Perhaps ensuring consistency with setField is one reason to use fieldNames rather than paths in getField variations [discussed above](https://github.com/mongodb/mongo-java-driver/pull/1052#discussion_r1036405352).